pkg/controller/generic_reconciler.go

```diff
@@ -193,11 +187,38 @@ func (gr *GenericReconciler) groupAppObjects(ctx context.Context,
 			}
 			listOptions.Continue = listContinue
 		}
-
+		relatedObjects = processOtherAppSelectors(relatedObjects, postProcessAppSelectors)
 	}
 	return relatedObjects, nil
 }
 
+// processOtherAppSelectors iterates over provided map of "app" label selectors and
+// it adds the corresponding object to the "relatedObjects" map based on the selector
+// operator value.
+func processOtherAppSelectors(
+	relatedObjects map[string][]*unstructured.Unstructured,
+	appSelectors map[*unstructured.Unstructured][]utils.AppSelector) map[string][]*unstructured.Unstructured {
+	for obj, appSelectors := range appSelectors {
+		for k := range relatedObjects {
+			for _, as := range appSelectors {
+				// if the selector operator is "Exists" then add it to every object
+				if as.Operator == metav1.LabelSelectorOpExists {
+					relatedObjects[k] = append(relatedObjects[k], obj)
+				}
+				// if the selector operator is "NotIn", then add it to every object
+				// except those that have a corresponding "app" label value.
+				if as.Operator == metav1.LabelSelectorOpNotIn {
+					if as.Values.Has(k) {
+						continue
+					}
+					relatedObjects[k] = append(relatedObjects[k], obj)
+				}
+			}
+		}
+	}
```

Comment:
This is pretty complex, but the assumption here is that the length of `relatedObjects` will be more often greater than length of `appSelectors`. Anothe possible impl would be:
```
	for obj, appSelectors := range postProcessAppSelector {
		for _, as := range appSelectors {
			if as.Operator == metav1.LabelSelectorOpExists {
				for k, _ := range relatedObjects {
					relatedObjects[k] = append(relatedObjects[k], obj)
				}
			} else if as.Operator == metav1.LabelSelectorOpNotIn {
				for k, _ := range relatedObjects {
					if as.Values.Has(k) {
						continue
					}
					relatedObjects[k] = append(relatedObjects[k], obj)
				}
			}
		}
	}
```
but this is iterating over the `relatedObjects` map multiple times. I can provide a benchmark test if anyone wants to experiment.